src/__tests__/WeekView.test.js

```diff
@@ -1,10 +1,454 @@
+import { render } from '@testing-library/react-native';
+import { State } from 'react-native-gesture-handler';
+import {
+  fireGestureHandler,
+  getByGestureTestId,
+} from 'react-native-gesture-handler/jest-utils';
 import WeekView from '../WeekView/WeekView';
 
-it('renders correctly', () => {
-  renderer.create(
-    <WeekView numberOfDays={3} selectedDate={new Date(2021, 3, 24)} />,
-  );
+/**
+ * Extracts the event-id embedded in "WeekViewEvent-<id>".
+ *
+ * Throws an error if the testID does not match the regex.
+ * @param {String} testID
+ * @returns string with event id
+ */
+const extractEventIdFromTestID = (testID) => {
+  const found = testID.match(/WeekViewEvent-(?<eventId>\d+)/);
+  const eventId = found && found.groups && found.groups.eventId;
+  if (!eventId) {
+    throw Error(`Could not find the event id in testID: ${testID}`);
+  }
+  return eventId;
+};
+
+/**
+ * NOTE: In most cases the props provided to <WeekView /> need to be
+ * selected carefully, so that events _are visible in the screen_.
+ * For example:
+ *   - numberOfDays
+ *   - selectedDate (usually, at least the same day the target event's startDate)
+ *   - hoursInDisplay
+ *   - startHour (usually, some hours before the target event)
+ */
+describe('Basic render functionality', () => {
+  describe('with no events', () => {
+    it('renders an empty grid', () => {
+      const { getByHintText, queryAllByHintText } = render(
+        <WeekView
+          events={[]}
+          numberOfDays={3}
+          selectedDate={new Date(2020, 3, 24)}
+        />,
+      );
+      expect(getByHintText('Grid with horizontal scroll')).toBeDefined();
+      expect(queryAllByHintText(/Show event \d+/)).toBeArrayOfSize(0);
+    });
+  });
+
+  const runTestShowsAll = (events, props) => {
+    const { queryAllByHintText, queryAllByText } = render(
+      /* eslint-disable-next-line react/jsx-props-no-spreading */
+      <WeekView events={events} {...props} />,
+    );
+    expect(queryAllByHintText(/Show event \d+/)).toBeArrayOfSize(events.length);
+    expect(queryAllByText(/event \d+/)).toBeArrayOfSize(events.length);
+
+    /**
+     * NOTE: Ideally, the selection would be only by text: queryAllByText('event \d')
+     * without queryAllByHintText(), but in practice an edge case might occur:
+     *     the box is visible --> events are detected byHint
+     *     but is too small, so the text is hidden --> events are not detected byText.
+     * This depends on many factors: screen size, hours in display, number of days,
+     * font size, padding, etc.
+     *
+     * To avoid future headaches with this test, we test both things in that order:
+     * byHint and byText.
+     *   - If byHint passes the test but byText does not
+     *     --> boxes are visible but the text inside is not
+     *     --> test events are too small, or change props hoursInDisplay, numberOfDays, etc
+     *   - If both test fails --> events are not rendered at all --> code is broken
+     */
```

Comment:
Yes it can happen in a test.

Since the size of the screen is constant in jest I'll remove `byHint`. The events in this test are large enough to be seen